# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/TabChild.cpp
# Commit: 365223fc9f86
# Full Hash: 365223fc9f865212c0f42948763ec48d4dca03ff
# Author: Barret Rennie <barret@brennie.ca>
# Date: 2019-04-06 10:57:50
# Description:
#   Bug 1542183 - Ensure TabChilds always have a TabGroup upon initialization r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D26374
# ==============================================================================

diff -r 21358d3572a7 -r 365223fc9f86 dom/ipc/TabChild.cpp
--- a/dom/ipc/TabChild.cpp	Fri Apr 05 19:25:30 2019 +0000
+++ b/dom/ipc/TabChild.cpp	Fri Apr 05 19:38:27 2019 +0000
@@ -507,9 +507,7 @@
 }
 
 nsresult TabChild::Init(mozIDOMWindowProxy* aParent) {
-  if (!mTabGroup) {
-    mTabGroup = TabGroup::GetFromActor(this);
-  }
+  MOZ_DIAGNOSTIC_ASSERT(mTabGroup);
 
   nsCOMPtr<nsIWidget> widget = nsIWidget::CreatePuppetWidget(this);
   mPuppetWidget = static_cast<PuppetWidget*>(widget.get());
